src/include/daos/mem.h

```diff
@@ -288,6 +288,13 @@ enum {
 	UMEM_TYPE_ANY,
 };
 
+/* Hints for umem atomic copy operation primarily for bmem implementation */
+enum acopy_hint {
+	UMEM_COMMIT_IMMEDIATE = 0, /* commit immediate, do not call within a tx */
+	UMEM_COMMIT_DEFER,	/* OK to defer commit to blob to a later point */
```

Comment:
I think the "defer" here means the reserved region will be committed to WAL by the associated transaction? Is that correct? Let take a typical use case as example:
1. Update calls reserve to reserve a region for value update;
2. Calls atomic_copy(UMEM_COMMIT_DEFER) to copy the value to reserved region;
3. Start local tx on vos_update_end(), and the reserved region _must_ have been added in WAL as copy_ptr entry, and the WAL must be committed on tx commit;

Is my assumption correct?

Reply:
UMEM_COMMIT_DEFER is used in places where atomic copy is called within a transaction.
UMEM_RESERVED_MEM is used for atomic copy on memory obtained from umem_reserve